packages/fs-detectors/test/unit.framework-detector-record.test.ts

```diff
@@ -0,0 +1,194 @@
+import frameworkList from '@vercel/frameworks';
+import { detectFrameworkRecord } from '../src';
+import VirtualFilesystem from './virtual-file-system';
+
+describe('detectFrameworkRecord', () => {
+  it('Do not detect anything', async () => {
+    const fs = new VirtualFilesystem({
+      'README.md': '# hi',
+      'api/cheese.js': 'export default (req, res) => res.end("cheese");',
+    });
+
+    const framework = await detectFrameworkRecord({ fs, frameworkList });
+    expect(framework?.slug).toBe(undefined);
+  });
+
+  it('Detects a framework record with a matchPackage detector', async () => {
+    const fs = new VirtualFilesystem({
+      'package.json': JSON.stringify({
+        dependencies: {
+          next: '9.0.0',
+        },
+      }),
+    });
+
+    const frameworkRecord = await detectFrameworkRecord({ fs, frameworkList });
+    if (!frameworkRecord) {
+      throw new Error(
+        '`frameworkRecord` was not detected, expected "nextjs" frameworks object'
+      );
+    }
+    expect(frameworkRecord.slug).toBe('nextjs');
+    expect(frameworkRecord.name).toBe('Next.js');
+    expect(frameworkRecord.detectedVersion).toBe('9.0.0');
+  });
+
+  it('Detects a framework record with a matchPackage detector with slashes', async () => {
+    const fs = new VirtualFilesystem({
+      'package.json': JSON.stringify({
+        dependencies: {
+          '@ionic/angular': '5.0.0',
+        },
+      }),
+    });
+
+    const frameworkRecord = await detectFrameworkRecord({ fs, frameworkList });
+    if (!frameworkRecord) {
+      throw new Error(
+        '`frameworkRecord` was not detected, expected "ionic-angular" frameworks object'
+      );
+    }
+    expect(frameworkRecord.slug).toBe('ionic-angular');
+    expect(frameworkRecord.detectedVersion).toBe('5.0.0');
+  });
+
+  it('Detect first framework version found', async () => {
+    const fs = new VirtualFilesystem({
+      'package.json': JSON.stringify({
+        dependencies: {
+          nuxt: '1.0.0',
+          nuxt3: '2.0.0',
+          'nuxt3-edge': '3.0.0',
+        },
+      }),
+    });
+
+    const framework = await detectFrameworkRecord({ fs, frameworkList });
+    expect(framework?.slug).toBe('nuxtjs');
+    expect(framework?.detectedVersion).toBe('1.0.0');
+  });
+
+  it('Detect frameworks based on ascending order in framework list', async () => {
+    const fs = new VirtualFilesystem({
+      'package.json': JSON.stringify({
+        dependencies: {
+          next: '9.0.0',
+          gatsby: '4.18.0',
+        },
+      }),
+    });
+
+    const framework = await detectFrameworkRecord({ fs, frameworkList });
+    expect(framework?.slug).toBe('nextjs');
+  });
```

Comment:
Another order related concern. Not that it is within scope of this PR, but this makes me think if theres a better way to do this. I'd assume we check in alphabetical order, but I know we intentionally stick `next.js` at the top of the list so that its "first".

Reply:
This is a test that `detectFrameworkRecord` works the same as `detectFramework` did. If we want to change how order works in the framework list, we should discuss that as a team separately, yeah.